archivebox/index/json.py

```diff
@@ -39,7 +39,20 @@
     },
 }
 
-### Main Links Index
+@enforce_types
+def generate_json_index_from_links(links: List[Link], with_headers: bool):
```

Comment:
this looks like a duplicate of `write_json_main_index`, do we need both?

Reply:
`write_json_main_index` has the sideeffect of writing to the filesystem. I think it can be safely removed in this version.+

Reply:
is the same true for `write_html_main_index`?

Reply:
Checking

Reply:
Nope, I don't think that is the case